### PR TITLE
proxy-agent: Use proxy-from-env and add docs

### PR DESCRIPTION
**Pull request by nhooyr**
Now we support pretty much every variable under the sun along with
`$NO_PROXY` all correctly and with minimal code on our end.

Also added an FAQ entry on them.

@code-asher I'd recommend looking at the PR diff instead of by commit as the early
commits are from my confusion but I'd like to include them for posterity.

### CODE REVIEW

**Review by code-asher — APPROVED** [review]

File: src/node/proxy_agent.ts [hunk 1]
```diff
@@ -15,48 +18,64 @@ import * as proxyagent from "proxy-agent"
...
-  if (!proxyURL) {
-    return
-  }
+export function monkeyPatch(inVSCode: boolean): void {
+  if (shouldEnableProxy()) {
+    const http = require("http")
+    const https = require("https")
 
-  logger.debug(`using $HTTP_PROXY ${process.env.HTTP_PROXY}`)
+    // If we do not pass in a proxy URL, proxy-agent will get the URL from the environment.
```
[the remark this answers is not in the document]
Comment (code-asher) [reply src/node/proxy_agent.ts:39]: I didn't put this on a good line, should've put it on the `if`. I was thinking about how it reads from the environment so I put it here. :stuck_out_tongue:

[the remark this answers is not in the document]
Comment (nhooyr) [reply src/node/proxy_agent.ts:39]: > Should we always set it? It looks like it does the same check except the downside is that it runs for every request.

I added the check for logging purposes. This is the log that uncovered the extension host logging bug for example.

[the remark this answers is not in the document]
Comment (nhooyr) [reply src/node/proxy_agent.ts:39]: logging/debugging

[the remark this answers is not in the document]
Comment (code-asher) [reply src/node/proxy_agent.ts:39]: Makes sense.

**Pull request merged**